numpy/core/tests/test_umath_accuracy.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+from os import path
+from ctypes import *
+from numpy.testing import assert_array_max_ulp
+
+# convert string to hex function taken from:
+# https://stackoverflow.com/questions/1592158/convert-hex-to-float #
+def convert(s):
+    i = int(s, 16)                   # convert from hex to a Python int
+    cp = pointer(c_int(i))           # make this into a c integer
+    fp = cast(cp, POINTER(c_float))  # cast the int pointer to a float pointer
+    return fp.contents.value         # dereference the pointer, get the float
+
+str_to_float = np.vectorize(convert)
+files = ['umath-validation-set-exp']
+
+class TestAccuracy(object):
+    def test_validate_transcendentals(self):
+        with np.errstate(all='ignore'):
+            for filename in files:
+                data_dir = path.join(path.dirname(__file__), 'data')
+                filepath = path.join(data_dir, filename)
+                data = np.genfromtxt(filepath,
+                                     dtype=('|S39','|S39','|S39',np.int),
```

Comment:
How about we make the validation file float32 specific? Then `convert` is just `lambda x: int(x, 16)`, the dtype starts off with `np.int16` and you can simply use `inval = arr["inval_hex"].view(np.float32)`. I think that is easier to read (in the context of numpy users, also makes the dtype less strange to understand).

Reply:
I was hoping to keep it generic for all dtypes, any particular reason to make it float32 specific?

Reply:
I just preferred the view method somewhat, and that does not work well with multiple types. But it is not a big thing, and we can always restructure things a bit.

Reply:
ok, I will leave it as it for now. We could re-factor later if needed.